examples/secured-service-with-ldap/secured_service_with_ldap.bal

```diff
@@ -0,0 +1,61 @@
+import ballerina/http;
+import ballerina/config;
+import ballerina/log;
+import ballerina/ldap;
+
+// Define LDAP connection configurations.
+ldap:LdapConnectionConfig ldapConfig = {
+    domainName: "ballerina.io",
+    connectionURL: "ldap://localhost:9095",
+    connectionName: "uid=admin,ou=system",
+    connectionPassword: "secret",
+    userSearchBase: "ou=Users,dc=ballerina,dc=io",
+    userEntryObjectClass: "identityPerson",
+    userNameAttribute: "uid",
+    userNameSearchFilter: "(&(objectClass=person)(uid=?))",
+    userNameListFilter: "(objectClass=person)",
+    groupSearchBase: ["ou=Groups,dc=ballerina,dc=io"],
+    groupEntryObjectClass: "groupOfNames",
+    groupNameAttribute: "cn",
+    groupNameSearchFilter: "(&(objectClass=groupOfNames)(cn=?))",
+    groupNameListFilter: "(objectClass=groupOfNames)",
+    membershipAttribute: "member",
+    userRolesCacheEnabled: true,
+    connectionPoolingEnabled: false,
+    connectionTimeoutInMillis: 5000,
+    readTimeoutInMillis: 60000,
+    retryAttempts: 3
+};
+
+// Creates an inbound LDAP authentication provider with the LDAP
+// connection configurations.
+ldap:InboundLdapAuthProvider ldapAuthProvider = new(ldapConfig, "ldap01");
+
+// Creates a Basic Auth handler with the created LDAP Auth provider.
+http:BasicAuthHandler ldapAuthHandler = new(ldapAuthProvider);
+
+// The endpoint used here is the `http:Listener`. The LDAP Auth handler is
+// set to this endpoint using the `authHandlers` attribute.
+// It is optional to override the authentication and authorization at the
+// service and resource levels.
+listener http:Listener ep = new (9090, {
+    auth: {
+        authHandlers: [ldapAuthHandler]
+    },
+    // The secure hello world sample uses HTTPS.
```

Comment:
```suggestion
    // The secure hello world sample uses HTTPS.
```
Do we need to link this to the BBE. Can you check if its possible to link?

Reply:
This lines describes on this BBE. It is actually a secure hello world sample.

Reply:
Can we link to whichever BBE that is applicable?

Reply:
I don't think we need a link here. It actually describe about the same BBE, not an external BBE.

Reply:
+1